packages/go/src/components/struct/declaration.tsx

```diff
@@ -0,0 +1,230 @@
+import {
+  Block,
+  Children,
+  computed,
+  Declaration,
+  DeclarationContext,
+  effect,
+  memo,
+  Namekey,
+  Refkey,
+  Scope,
+  Show,
+  takeSymbols,
+  untrack,
+  useContext,
+  watch,
+} from "@alloy-js/core";
+import { useGoScope, useNamedTypeScope } from "../../scopes/contexts.js";
+import { createNamedTypeScope } from "../../scopes/factories.js";
+import { GoNamedTypeScope } from "../../scopes/named-type.js";
+import {
+  createAnonymousTypeSymbol,
+  createStructMemberSymbol,
+} from "../../symbols/factories.js";
+import { GoSymbol, isNameExported } from "../../symbols/go.js";
+import { NamedTypeSymbol } from "../../symbols/named-type.js";
+import { LineComment } from "../doc/comment.js";
+import { Name } from "../Name.js";
+import { TypeDeclaration, TypeDeclarationProps } from "../type/declaration.jsx";
+
+// properties for creating a struct
+export interface StructDeclarationProps {
+  refkey?: Refkey;
+  children?: Children;
+
+  /**
+   * Whether to render the struct in a single line.
+   * This will only compile if `children` is a single line as well.
+   * This is not common in Go, so use with caution.
+   */
+  singleLine?: boolean;
+}
+
+/**
+ * Wrapper for creating a named struct type declaration.
+ */
+export function StructTypeDeclaration(
+  props: StructDeclarationProps & TypeDeclarationProps,
```

Comment:
Let's use interface extends here (it will work better with the documentation system).